### PR TITLE
Clean up changelog

### PR DESCRIPTION
**Pull request by ru-fu**
Consistent spelling for "changelog" and some fixes for formatting issues.

### CODE REVIEW

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -64,12 +67,12 @@ The following list summarizes the most important changes inherited from upstream
 
   * MQTT:
 
-    * The ``utf8`` pointer in the :c:type:`mqtt_utf8` struct is now const.
+    * The ``utf8`` pointer in the :cpp:class:`mqtt_utf8` struct is now const.
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:70]: This is not a C++ class, it's a C struct. If this needs to be changed, `:c:struct:` would be more appropriate. Is this being copy/pasted from elsewhere in the documentation? If so, further cleanup may be worth doing.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-c-struct

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -93,13 +96,13 @@ The following list summarizes the most important changes inherited from upstream
   * Deprecated ``BT_LE_SCAN_FILTER_WHITELIST``.
     Use :cpp:enumerator:`BT_LE_SCAN_OPT_FILTER_WHITELIST <bt_gap::BT_LE_SCAN_OPT_FILTER_WHITELIST>` instead.
   * Deprecated ``bt_le_scan_param::filter_dup``.
-    Use ``bt_le_scan_param::options`` instead.
+    Use :cpp:member:`bt_le_scan_param::options` instead.
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:99]: Similarly, this is not a C++ entity; maybe `:c:member:` if this needs changing.

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -93,13 +96,13 @@ The following list summarizes the most important changes inherited from upstream
...
   * Deprecated ``bt_conn_create_le()``.
     Use :cpp:func:`bt_conn_le_create` instead.
   * Deprecated ``bt_conn_create_auto_le()``.
-     Use :cpp:func:`bt_conn_le_create_auto` instead.
+    Use :cpp:func:`bt_conn_le_create_auto` instead.
   * Deprecated ``bt_conn_create_slave_le()``.
-    Use :cpp:func:`bt_le_adv_start` instead, with ``bt_le_adv_param::peer`` set to the remote peer's address.
+    Use :cpp:func:`bt_le_adv_start` instead, with :cpp:member:`bt_le_adv_param::peer` set to the remote peer's address.
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:105]: Ditto here, although now I see that various things are being documented in the C++ domain which should really be in the C domain.

File: doc/nrf/releases/release-notes-latest.rst (hunk 1 above)
Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:70]: This is a limitation in Breathe. When using :c:type:, the link will not work. That's why we need to use :cpp:class: instead - @utzig , can you elaborate?

File: doc/nrf/releases/release-notes-latest.rst (hunk 3 above)
Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:105]: Same here - this is the only way to get the links working.

File: doc/nrf/releases/release-notes-latest.rst (hunk 1 above)
Comment (utzig) [reply doc/nrf/releases/release-notes-latest.rst:70]: > This is not a C++ class, it's a C struct. If this needs to be changed, `:c:struct:` would be more appropriate. Is this being copy/pasted from elsewhere in the documentation? If so, further cleanup may be worth doing.
> 
> https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-c-struct

This was introduced with Sphinx>3 and we are still using `2.4.4`. Also breathe parsed most Doxygen XML as if it was cpp until very recently. Using `:cpp:class`  is still the correct way to do the linking (as you must be quite used to seeing `:cpp:func`) until we upgrade to latest tooling.

Comment (mbolivar-nordic) [reply doc/nrf/releases/release-notes-latest.rst:70]: > as you must be quite used to seeing `:cpp:func`

I'm not used to this. For example `:c:func:` is used to link to devicetree macros throughout zephyr/doc/guides/dts/api-usage.rst. These work.

Grepping `zephyr/doc` for `:c:func:` has 668 matches total, many of which are to legitimate functions, not just macros. However, it looks like you're right that when linking to functions, these don't generate links.

This seems like a bug which should be fixed in the upstream zephyr docs, I guess. Or perhaps we can upgrade our tooling so the C domain works; using C++ sounds like it's a workaround.

Anyway, thanks for explaining, seems fine for now!

**Pull request merged**